aten/src/ATen/nn_parse.py

```diff
@@ -418,12 +415,22 @@ def run(paths):
                 if cname + suffix in header_functions:
                     bwd_functions.append(header_functions[cname + suffix])
 
-            base = base_declaration(func, fwd_function, backends)
-            declarations.append(forward_declaration(base, fwd_function))
-            declarations.append(backward_declaration(base, bwd_functions))
+            default_scalar_types = ['Float', 'Double', 'Half']  # Half will be stripped for CPU backend
+            forward_backend_types = {}
+            backward_backend_types = {}
+            for backend in backends:
+                backend_props = func.get(backend, {})
+                forward_backend_types[backend] = backend_props.get('forward_scalar_types', default_scalar_types)
+                backward_backend_types[backend] = backend_props.get('backward_scalar_types', default_scalar_types)
+
+            base = base_declaration(func, fwd_function, backends, default_scalar_types)
```

Comment:
why is this default_scalar_types instead of forward_scalar_types?

Reply:
'base' will be used as default and it will populated to forward_declaration and backward_declaration function with forward_backend_types and backward_backend_types to be able to override base's default types.

Reply:
discussed in person, going to try to use `None` to indicate that this value doesn't matter / isn't used.

Reply:
Done